src/libraries/Microsoft.Extensions.Http/src/Logging/LoggingHttpMessageHandler.cs

```diff
@@ -144,11 +147,25 @@ public static void RequestEnd(ILogger logger, HttpResponseMessage response, Time
                 }
             }
 
-            private static string? GetUriString(Uri? requestUri)
+            internal static string? GetUriString(Uri? uri)
             {
-                return requestUri?.IsAbsoluteUri == true
-                    ? requestUri.AbsoluteUri
-                    : requestUri?.ToString();
+                if (uri is null)
+                {
+                    return null;
+                }
+
+                if (uri.IsAbsoluteUri)
+                {
+                    return s_logQueryString
+                        ? uri.AbsoluteUri
+                        : uri.GetComponents(UriComponents.AbsoluteUri & ~UriComponents.Query, UriFormat.UriEscaped);
+                }
+
+                string uriString = uri.ToString();
+                int queryOffset = s_logQueryString ? -1 : uriString.IndexOf('?');
+                return queryOffset < 0
+                    ? uriString
+                    : uriString.Substring(0, queryOffset);
```

Comment:
I've been thinking that we might want to leave a "hint" that we've redacted something here. dotnet/extensions logging uses `REDACTED`; but our own header redaction in HttpClientFactory substitutes header values with `*` -- so I've been thinking that we need to align and also substitute the query with `*`?

Also given the headers are not removed, but substituted instead, it also seems more suitable to substitute (instead of removing)